stdlib/asm/crypto/hashes/rpo.masm

```diff
@@ -54,8 +54,8 @@ end
 #! Input: [start_addr, end_addr, ...]
 #! Output: [H, ...]
 #! Cycles:
-#! even words: 48 cycles + 3 * words
-#! odd words: 60 cycles + 3 * words
+#! even words: 49 cycles + 3 * words
+#! odd words: 61 cycles + 3 * words
 export.hash_memory
```

Comment:
One thing that I realized: I'm not sure the behavior of this procedure is entirely consistent with its naming/docs. Specifically, we assume that the number of elements to be hashed is a multiple of 4 - so, for example, it cannot be used to hash 9 elements.

Maybe a better approach would be to split it into 2 procedures:
```
#! Input:  [C, B, A, start_addr, num_words, ...]
#! Output: [C', B', A', end_addr, ...]
export.absorb_words_from_memory
```
The above procedure would be similar to `absorb_double_words_from_memory` but would handle the case of the odd number of words. (separately, I think we could modify the `absorb_double_words_from_memory` procedure to also take `num_words` instead of `end_addr`).

The second procedure would actually be for computing hash of memory:
```
#! Input:  [start_addr, num_elements, ...]
#! Output: [HASH, end_addr, ...]
export.hash_memory
```
One open question is whether we should assume that the last word is padded with zeros (in case the number of elements is not divisible by 4) or if we need to handle this in the procedure.